ci/tasks/scripts/with-docker.sh

```diff
@@ -2,19 +2,28 @@
 
 set -e -u -o pipefail
 
+# Launches the docker daemon, and loads our cached images from the `docker-cache` folder.
+
 source /docker-helpers.sh
 start_docker
 
 function load_layers() {
+
+  # don't do anything if we don't have cache
+  [[ ! -d 'docker-cache' ]] && return 0;
+  [[ -z "$(ls docker-cache/*/image.tar)" ]] && return 0;
+
   echo "loading docker layer cache"
   pids=
-  docker load -qi mysql-image/image.tar & pids[0]=$!
-  docker load -qi nginx-image/image.tar & pids[1]=$!
-  docker load -qi ruby-image/image.tar & pids[2]=$!
-  [[ -f "govwifi-admin-prebuilt/image.tar" ]] && docker load -qi "govwifi-admin-prebuilt/image.tar" & pids[3]=$!
+
+  index=0
+  for cached_image in docker-cache/*/image.tar; do
+    docker load -qi "${cached_image}" & pids[${index}]=$!
+    index="$(( "${index}" + 1 ))"
+  done
 
   for pid in ${pids[*]}; do
-    wait "$pid" || true  # we don't care about the return code of docker loading
+    wait "$pid"
```

Comment:
Just out of curiosity, we now care about the return code of docker loading?

Reply:
ah, explanation for this one.

This occurred when the cache wasn't populated, and relates to the old line 14.

```
[[ -f "govwifi-admin-prebuilt/image.tar" ]] && docker load -qi "govwifi-admin-prebuilt/image.tar" & pids[3]=$!
```

The `& pids[3]=$!` is run on the whole statement, not just the part after the condition check.
This means that for when the file doesn't exist, the process of that statement will return an exit code of 1 due to the `[[ ... ]]` returning falsely.

adding `wait || true` was a hack to work around the issue as I didn't fully understand the cause before.

With the move to globbing, the only issues we'll get are ones we want to raise